pyodide_build/pywasmcross.py

```diff
@@ -262,6 +268,13 @@ def handle_command(line, args, dryrun=False):
         # Don't include any system directories
         if arg.startswith("-L/usr"):
             continue
+        skip_lib = False
```

Comment:
I've looked into this more, and I noticed there are three problems arising from using emscripten ports (`-s USE_FOO=1`)
1. Before emscripten 1.39.5, the linker doesn't know where to look for these ports, and so specifying `-lfoo` gives an error. This results in what's at line 323. This can be removed after upgrading emscripten.
2. `-lfoo` looks for `.a` files but we have `.bc` files for some (but not all) ports. This explains why we don't have to supply extra ldflags for `freetype` and `zlib`, but we need it for `libpng`. With the upstream backend, all ports are supplied in the form of `.a`, so we are good here.
3. In matplotlib, it supplies `-lpng16`, but we only have `libpng.a`, not `libpng16.a`. In this case the right course of action is to *replace* `-lpng16` with `-lpng`, not ignore it all together. This problem is common to both 1.38.44+fastcomp and 2.0.x+upstream

So I think, ultimately, there are no libraries that we ought to skip, but instead we should supply replacements

Reply:
aha, so we could instead of having ignore libraries, have replace-libraries as a flag? So one could replace-libraries png16=png;someotherlibrary.3.2=someotherlibrary or to ignore a library, could just do replace-libraries: gfortran=;

On fastcomp I guess it ought to do something in pywasmcross to fix the bc file problem so -lpng works?

I'll put the different things in different patches in a mo.

Reply:
I think before we fully commit to upstream backend, just having the replace-library flag would do; we have "hacks" in place for the other problems